programming_challenges/hacker_rank/filter-array.clj

```diff
@@ -1,10 +1,9 @@
 ;; https://www.hackerrank.com/challenges/fp-filter-array/problem
 
 (defn filter-array
-  [n list]
-  (if (< (first list) n)
-    (println (first list)))
-  (if (not (empty? (rest list)))
-    (filter-array n (rest list))))
+  [n lst]
+  (filter
+    (fn [list-element] (< list-element n))
```

Comment:
Usually `filter` and `map` use their aux fnx as a new function instead an anonymus fn.

Reply:
@marinalimeira In this case, how do I create an auxiliar function for `filter`?

I implemented the aux function

```clojure
(defn greater?
  [list-element n]
  (< list-element n))
```

And I tried something like this

```clojure
(defn filter-array
  [n lst]
  (filter
    (greater? list-element n)
    lst))
```

But I need the `list-element` to pass as parameter to the aux function.

Another approach was to build an anonymous function to have the `list-element`

```clojure
(defn filter-array
  [n lst]
  (filter
    (fn [list-element] (greater? list-element n))
    lst))
```

Do you know another approach to this problem?